crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -264,6 +302,16 @@ impl<'a> ReflectDerive<'a> {
 
         let meta = ReflectMeta::new(type_path, traits);
 
+        if provenance.source == ReflectImplSource::ImplRemoteType
+            && meta.traits.type_path_attrs().should_auto_derive()
```

Comment:
Nit: On main this can be simplified
```suggestion
            && meta.type_path_attrs().should_auto_derive()
```